internal/pkg/cli/svc_init.go

```diff
@@ -476,11 +476,18 @@ func legitimizePlatform(engine dockerEngine, wkldType string) (manifest.Platform
 	if redirectedPlatform == "" {
 		return "", nil
 	}
-	if redirectedPlatform != detectedPlatform && wkldType != manifest.RequestDrivenWebServiceType {
-		log.Warningf("Your architecture type %s is currently unsupported. Setting platform %s instead.\n", color.HighlightCode(detectedArch), redirectedPlatform)
+	// Return an error if a platform cannot be redirected.
+	if wkldType == manifest.RequestDrivenWebServiceType && detectedOs == manifest.OSWindows {
+		return "", manifest.ErrAppRunnerInvalidPlatformWindows
+	}
+	if detectedArch == manifest.ArchARM || detectedArch == manifest.ArchARM64 {
+		if wkldType != manifest.RequestDrivenWebServiceType {
+			log.Warningf("Architecture type %s has been detected. We will set platform '%s' instead. If you'd rather build and run as architecture type %s, please change the 'platform' field in your workload manifest to '%s'.\n", detectedArch, dockerengine.PlatformString(detectedOs, manifest.ArchAMD64), manifest.ArchARM64, detectedPlatform)
+		} else {
+			log.Warningf("Architecture type %s has been detected. At this time, %s architectures are not supported for App Runner workloads. We will set platform '%s' instead.\n", detectedArch, detectedArch, dockerengine.PlatformString(detectedOs, manifest.ArchAMD64))
```

Comment:
this is really a nitpick, but one way I like removing `else` statements is to refactor like this:
```go
msg := fmt.Sprintf("Architecture type %s has been detected. We will set platform '%s' instead. If you'd rather build and run as architecture type %s, please change the 'platform' field in your workload manifest to '%s'.\n", detectedArch, dockerengine.PlatformString(detectedOs, manifest.ArchAMD64), manifest.ArchARM64, detectedPlatform)
if wkldType == manifest.RequestDrivenWebService {
   msg = fmt.Sprintf("Architecture type %s has been detected. At this time, %s architectures are not supported for App Runner workloads. We will set platform '%s' instead.\n", detectedArch, detectedArch, dockerengine.PlatformString(detectedOs, manifest.ArchAMD64))
}
log.Warningf(msg)
```
So the common scenario is moved outside of a conditional, and the conditional is only for the rarer-cases.

Reply:
Should it be `redirectedPlatform` instead of `dockerengine.PlatformString(detectedOs, manifest.ArchAMD64)` in the log to be safe?